components/feature/prompts/src/main/java/mozilla/components/feature/prompts/address/AddressPicker.kt

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.address
+
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.prompt.PromptRequest
+import mozilla.components.concept.storage.Address
+import mozilla.components.feature.prompts.concept.SelectablePromptView
+import mozilla.components.feature.prompts.consumePromptFrom
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * Interactor that implements [SelectablePromptView.Listener] and notifies the feature about actions
+ * the user performed in the address picker.
+ *
+ * @property store The [BrowserStore] this feature should subscribe to.
+ * @property addressSelectBar The [SelectablePromptView] view into which the select address
+ * prompt will be inflated.
+ * @property onManageAddresses Callback invoked when user clicks on "Manage adresses" button from
+ * select address prompt.
+ * @property sessionId The session ID which requested the prompt.
+ */
+class AddressPicker(
+    private val store: BrowserStore,
+    private val addressSelectBar: SelectablePromptView<Address>,
+    private val onManageAddresses: () -> Unit = {},
+    private var sessionId: String? = null
+) : SelectablePromptView.Listener<Address> {
+
+    init {
+        addressSelectBar.listener = this
+    }
+
+    /**
+     * Shows the select address prompt in response to the [PromptRequest] event.
+     *
+     * @param request The [PromptRequest] containing the the address request data to be shown.
+     */
+    internal fun handleSelectAddressRequest(request: PromptRequest.SelectAddress) {
+        addressSelectBar.showPrompt(request.addresses)
+    }
+
+    /**
+     * Dismisses the active [PromptRequest.SelectAddress] request.
+     *
+     * @param promptRequest The current active [PromptRequest.SelectAddress] or null
+     * otherwise.
+     */
+    @Suppress("TooGenericExceptionCaught")
+    fun dismissSelectAddressRequest(promptRequest: PromptRequest.SelectAddress? = null) {
+        addressSelectBar.hidePrompt()
+
+        try {
+            if (promptRequest != null) {
+                promptRequest.onDismiss()
+                return
```

Comment:
Is there any specific reason why we are not also consing the `promptRequest` here ? :)

Reply:
Below [in line 61](https://github.com/mozilla-mobile/android-components/pull/12142/files#diff-6370c01c1105bce2bcc2e77790378bef39be42d8babf6d27dac38d4e67ce3317R61) we are doing it, but here we are not, I wonder why? :)

Reply:
Normally we always want to consume the `promptRequest` after finish using it, as it clears the store reference.

Reply:
Great catch, do you think this should also be addressed in `dismissSelectCreditCardRequest` and `dismissCurrentLoginSelect`, since they follow the same logic?

Reply:
Let's address it a separate ticket

Reply:
Opened https://github.com/mozilla-mobile/android-components/issues/12241 for `dismissSelectCreditCardRequest` and `dismissCurrentLoginSelect`.

Reply:
Thanks!